### PR TITLE
DOC: fix rst link syntax in changelog

### PR DESCRIPTION
**Pull request by GaelVaroquaux**
Fix a trivial formating bug.

### CODE REVIEW

**Review by lucyleeow — COMMENTED** [review]
Thanks! Just a query

File: CHANGES.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@ Next version
 
 In this version, the "Out:" prefix applied to code outputs is now created from
 CSS pseudo-elements instead of additional real text. For more details, see
-`#896 <https://github.com/sphinx-gallery/sphinx-gallery/pull/896>`.
+`#896 <https://github.com/sphinx-gallery/sphinx-gallery/pull/896>`__.
```
[the remark this answers is not in the document]
Comment (larsoner) [reply CHANGES.rst:9]: The double-`_` makes it an anonymous link, which I think is fine (probably better actually) here

**Comment by larsoner** [comment]
Thanks @GaelVaroquaux !

File: CHANGES.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucyleeow) [reply CHANGES.rst:9]: I wish I knew this before! It would have so useful

**Pull request merged**